README.md

```diff
@@ -21,7 +21,31 @@ Numaplane is a control plane for installing, managing and running numaflow resou
 ## Contributing
 **NOTE:** Run `make --help` for more information on all potential `make` targets
 
-More information can be found via the [Kubebuilder Documentation](https://book.kubebuilder.io/introduction.html)
+## How To Release
+
+### Release Branch
+
+Always create a release branch for the releases, for example branch `release-0.5` is for all the v0.5.x versions release.
+If it's a new release branch, simply create a branch from `main`.
+
+### Release Steps
+
+1. Cherry-pick fixes to the release branch, skip this step if it's the first release in the branch.
+2. Run `make test` to make sure all test cases pass locally.
+3. Push to remote branch, and make sure all the CI jobs pass.
```

Comment:
I wonder if we really need to do steps 2 and 3 if we didn't cherry pick in step 1?

Reply:
Yeah I don't think it is needed